tests-ng/test_users.py

```diff
@@ -1,13 +1,14 @@
 import os
 import pwd
+from typing import Set
 import pytest
 import stat
 
-def test_service_accounts_have_nologin_shell():
+def test_service_accounts_have_nologin_shell(expected_users: Set[str]):
     for entry in pwd.getpwall():
         if entry.pw_uid >= 1000:
             continue
-        if entry.pw_name in {"root", "sync"}:
+        if entry.pw_name in expected_users | {"root", "sync"}:
```

Comment:
Even for expected users, if they are service accounts (UID < 1000) they should NOT have a login shell. For regular users with UID starting at 1000 this is skipped anyways.

Reply:
If lima uses UID 501 for regular users, then we should probably set `UID_MIN` in `/etc/login.defs` accordingly (in the lima feature) and modify this test to use that value dynamically as the range of system users rather than hardcoded 1000